packages/devtools_app/lib/src/ui/hover.dart

```diff
@@ -215,23 +223,93 @@ class HoverCard {
   }
 }
 
+/// Ensures that only one [HoverCard] is ever displayed at a time.
+class HoverCardTooltipController {
+  /// The card that is currenty being displayed
+  HoverCard? _currentHoverCard;
+
+  /// Sets [hoverCard] as the most recently displayed [HoverCard].
+  ///
+  /// This methods sets a unique [hoverCard.freshnessToken], as a way to track whether
+  /// [hoverCard] is the most recently displayed [HoverCard].
+  void set({required HoverCard hoverCard}) {
+    _currentHoverCard?.remove();
+    _currentHoverCard = hoverCard;
+  }
+
+  /// If the mouse is outside of [_currentHoverCard] then then it will be removed.
+  void maybeRemoveHoverCard(HoverCard hoverCard) {
+    if (isHoverCardStillActive(hoverCard)) {
+      final wasRemoved = _currentHoverCard?.maybeRemove();
+      if (wasRemoved == true) {
+        _currentHoverCard = null;
+      }
+    }
+  }
+
+  /// Remove [hoverCard] if it is currently active.
+  void removeHoverCard(HoverCard hoverCard) {
+    if (isHoverCardStillActive(hoverCard)) {
+      _currentHoverCard?.remove();
+      _currentHoverCard = null;
+    }
+  }
+
+  /// Checks if the [hoverCard] is still the active [HoverCard].
+  bool isHoverCardStillActive(HoverCard hoverCard) {
+    return _currentHoverCard == hoverCard;
+  }
+}
+
 /// A hover card based tooltip.
 class HoverCardTooltip extends StatefulWidget {
+  /// A [HoverCardTooltip] that generates it's [HoverCardData] asynchronously.
+  ///
+  /// [asyncGenerateHoverCardData] is used to generate the data that will
+  /// display in the final [HoverCard]. While that data is being generated,
+  /// a [HoverCard] with a spinner will show. If any [HoverCardData] returned
+  /// from [asyncGenerateHoverCardData] the spinner [HoverCard] will be replaced
+  /// with one containing the generated [HoverCardData].
   const HoverCardTooltip({
```

Comment:
maybe we should make this HoverCardTooltip.async so it is immediately clear which constructor to use based on your use case?